rook/migration/v1.14/README.md

```diff
@@ -0,0 +1,233 @@
+replaced with updated versions in sequence. The cluster may be unresponsive very briefly as mons update,
+and the Ceph Filesystem may fall offline a few times while the MDSes are upgrading. This is normal.
+
+The versions of the components can be viewed as they are updated:
+
+```console
+watch --exec kubectl -n $ROOK_CLUSTER_NAMESPACE get deployments -l rook_cluster=$ROOK_CLUSTER_NAMESPACE -o jsonpath='{range .items[*]}{.metadata.name}{"  \treq/upd/avl: "}{.spec.replicas}{"/"}{.status.updatedReplicas}{"/"}{.status.readyReplicas}{"  \trook-version="}{.metadata.labels.rook-version}{"\n"}{end}'
+```
+
+As an example, this cluster is midway through updating the OSDs. When all deployments report `1/1/1`
+availability and `rook-version=v1.14.2`, the Ceph cluster's core components are fully updated.
+
+```console
+Every 2.0s: kubectl -n rook-ceph get deployment -o j...
+
+rook-ceph-mgr-a         req/upd/avl: 1/1/1      rook-version=v1.14.2
+rook-ceph-mon-a         req/upd/avl: 1/1/1      rook-version=v1.14.2
+rook-ceph-mon-b         req/upd/avl: 1/1/1      rook-version=v1.14.2
+rook-ceph-mon-c         req/upd/avl: 1/1/1      rook-version=v1.14.2
+rook-ceph-osd-0         req/upd/avl: 1//        rook-version=v1.14.2
+rook-ceph-osd-1         req/upd/avl: 1/1/1      rook-version=v1.13.8
+rook-ceph-osd-2         req/upd/avl: 1/1/1      rook-version=v1.13.8
+```
+
+An easy check to see if the upgrade is totally finished is to check that there is only one
+`rook-version` reported across the cluster.
+
+```console
+# kubectl -n $ROOK_CLUSTER_NAMESPACE get deployment -l rook_cluster=$ROOK_CLUSTER_NAMESPACE -o jsonpath='{range .items[*]}{"rook-version="}{.metadata.labels.rook-version}{"\n"}{end}' | sort | uniq
+This cluster is not yet finished:
+  rook-version=v1.13.8
+  rook-version=v1.14.2
+This cluster is finished:
+  rook-version=v1.14.2
+```
+
+### **5. Verify the updated cluster**
+
+At this point, the Rook operator should be running version `rook/ceph:v1.14.2`.
+
+Verify the CephCluster health using the [health verification doc](health-verification.md).
+
+### **6. Disable holder pods**
+
+Rook is beginning the process of deprecating CSI network "holder" pods. If there are pods named
+`csi-*plugin-holder-*` in the Rook operator namespace, see the
+[detailed documentation](../CRDs/Cluster/network-providers.md#holder-pod-deprecation)
+to disable them. This is optional for v1.14, but will be required in a future release.
+    ```
+
+## Upgrade Ceph Reef
+
+### **1. Update the Ceph daemons**
+
+The upgrade will be automated by the Rook operator after the desired Ceph image is changed in the
+CephCluster CRD (`spec.cephVersion.image`).
+
+```console
+ROOK_CLUSTER_NAMESPACE=rook-ceph
+NEW_CEPH_IMAGE='quay.io/ceph/ceph:v18.2.2-20240311'
+kubectl -n $ROOK_CLUSTER_NAMESPACE patch CephCluster $ROOK_CLUSTER_NAMESPACE --type=merge -p "{\"spec\": {\"cephVersion\": {\"image\": \"$NEW_CEPH_IMAGE\"}}}"
+```
```

Comment:
These migration all feel copy-pasted from somewhere else and are not applicable to how we deploy rook-ceph.